### PR TITLE
Mash code

### PR DESCRIPTION
**Pull request by pflanzenhueter**

### CODE REVIEW

File: MASH-FRET/source/traces/processing/cross-talk/gammaCorr.m [hunk 1]
```diff
@@ -17,23 +17,44 @@
     % collect molecule traces
     I_den = p.proj{proj}.intensities_denoise(:, ...
         ((mol-1)*nChan+1):mol*nChan,:);
+    I_DTA = p.proj{proj}.intensities_DTA(:, ...
+        ((mol-1)*nChan+1):mol*nChan,:);
+    
+    % set method to "manual" if no intensity discretization is found
+    if sum(isnan(I_DTA))
```
Comment (fdsteffen) [inline MASH-FRET/source/traces/processing/cross-talk/gammaCorr.m]: Currently the "if" statement returns a non-zero-value whenever NaNs are found (i.e. when the trace is not discretized at all) and thus resets to "manual". But if the traces is only cut, then those values are also NaNs. Therefore I suggest to change the line to 
```
all(isnan(I_DTA))
```
This checks if all values are NaNs (i.e. no discretization of the intensity trace)

Comment (mca-sh) [reply MASH-FRET/source/traces/processing/cross-talk/gammaCorr.m]: I agree, thank you for the suggestion!

**Comment by mca-sh** [comment]
Here is a release note:

# Requirements:
- delete `MASH-FRET\default_param.ini` before starting MASH the first time

# Fixed bugs:

### General

- **fix trace file restructure error** when MATLAB's older than 2016b was used 
- **correct trace file restructure** for multiple FRET and S data (FRET and S import usable in TA only)

### Simulation

- **correct TIRF background calculation** by including background intensities as Gaussian amplitudes

### Video processing

- **fix coordinates import error** when loading coordinates for intensity integration
- **fix set project options error** when FRET or S calculation are not possible with given experiment settings
- **correct colormap in Mapping tool** back to old MATLAB's default `jet` for greater contrast
- **correct set project options** by updating channel menus for FRET and S calculations after modifying channel's label

### Trace processing

- **fix gamma correction error** when photobleaching-based calculations do not converge
- **fix coordinates channel list error** when loading a project with fewer channels than last project
- **fix coordinates import error** when importing coordinates located in an external ASCII file
- **fix trace discretization error** when all ratio data are out of the `[-0.2,1.2]` range
- **fix discretization parameter error** when setting the third parameter
- **fix edit project options error** when the number of FRET and S calculations was downsized for a same project
- **fix edit project options error** when FRET or S calculation are not possible with given experiment settings
- **correct molecule assignment** of discretized FRET and stoichiometry traces after clearing a selection of molecules from the list
- **correct gamma calculation**using discretized intensities and not from noisy intensities (as before my modifications)
- **correct gamma file export and import ** for multiple donor-acceptor FRET pair
- **correct coordinates import failure management** when importing coordinates located in an external ASCII file
- **correct parameters saving and import** when modifying import options for coordinates in import options for ASCII traces
- **correct trace top-discretization** by using the tolerance window parameter defined for the concerned bottom trace
- **correct cross-talk subtraction** in two steps by first, subtracting *bt* from all intensities and second, subtracting *DE*
- **correct processing parameter file** with correct molecule index and correct method for photobleaching detection
- **correct edit project options** by updating default label list with imported project's labels, and updating channel menus for FRET and S calculations after modifying channel's label
- **correct import of ASCII traces** with different length by adapting the `bool_intensities` project's properties and automatically cut missing data from traces
- **correct plot in bottom axes** when several stoichiometry and FRET traces are available by selecting the correct data

### Histogram analysis

- **correct GUI update** of BOBA checkbox value

### Transition analysis

- **fix TDP export error** when exporting ASCII data
- **fix clustering error** when activating plot of EM-results (internal use only)
- **correct checking of k-mean starting guess** and start clustering if all states are different
- **correct import of ASCII traces** when importing discretized FRET and adapt to multiple donor-acceptor FRET pairs
  
# Improvements:

### General
- *review saving of MASH projects* by updating current project's properties with new destination file, dates and MASH versions (imitate a "Save to...")

### Simulation
- **change GUI nomenclature** of noise model `none` to `Offset only` and background type `Constant` to `Uniform`
- **reduce computation time for NExpN noise generation** from 4h to 20min by restricting input image counts data to integers

### Video processing

- **change project's default destination** back to root folder
- **remove double warning** when coordinates import for intensity integration fails
- **improve coordinates import** for intensity integration with coordinates written in two columns only (as in *.map file)
- **change GUI nomenclature** in panel "Edit and export videos" of filter names according to doc
- **manage common user mistakes* in definition of project options
- **set stable data default colors** for intensities to green, orange and red

### Trace processing

- **cancel systematic drop of molecule list** when updated (after setting molecule label, molecule selection, clearing molecules from list, update project list etc.)
- **reset gamma calculation method to manual** when photobleaching-based calculations did not converge (as before my modifications)
- **leave DE coefficient visible** when calculations are not possible
- **Make gamma factor section's title invisible** when calculation are not possible
- **display the whole list of errors** when `intAscii2mash.m` execution fails
- **change gamma file import system** when importing ASCII traces, from automatically asked to defined in ASCII import options
- **review ASCII import options GUI** by separating the "discretized FRET" option into a "State trajectory" panel (will be updated with more options later)
- **review Background analyzer GUI** with data selection as in Background correction panel and more compact interface
- **review cross-talk parameters structure** by reducing *bt* coefficients dimensions to reduced to `[nChan,nChan-1]` and *DE* coefficient to `[nExc-1,nChan]`
- **review panel Factor corrections GUI** with new data selection for cross-talk coefficient settings and nomenclature adapted from module Simulation
- **improve processing parameter file** by sorting content into sections "PROJECT", "VIDEO PROCESSING", "EXPERIMENT SETTINGS" and "MOLECULE", replacing channel indexes (1,2,etc.) by emitter labels (Cy3,Cy5,etc.) and adding parameters "date of export", "MASH version at project creation" and "molecule tag"
- **improve panel "Find states" GUI** by adapting the data list to the chosen discretization mode, and the parameters 'Enable' property to the selected data.
- **manage common user mistakes* in project options editing
- **set default discretization method** to STaSI to get reliable state trajectories at opening
- **offer option import gamma from file** in "Factor corrections" panel by changing the "Opt." button to "Load" if method is manual
- **set stable data default colors** for intensities to green, orange and red

### Histogram analysis

- **set default FRET histogram bin** to `-0.2:0.025:1.2`

# New functionalities

### General

- **create side function to generate FRET expression** as used in MASH for multi-color FRET experiments and make it available at `MASH-FRET/buildFretExpr.m`
- **make NExpN distribution fitting function available** at `MASH-FRET/fit_NExpN.m`

**Comment by pflanzenhueter** [comment]
I very much like your way in commenting the pull-request. It is much more easy for me to review then. 

Thanks.

Ps. nothing to add. I will only accept.

**Pull request merged**